### DISCUSSION TITLE
Bad request  INVALID_IMAGE_OPTIMIZE_REQUEST when using the Image component with S3

### DISCUSSION THREAD

**Discussion by sballesteros** [vercel/next.js#20138] [Help]
Hello, not sure what I am doing wrong here:

next-config.js:
```js
  images: {
    domains: ['bucket.region.amazonaws.com']
  }
```

JSX:
```html
<Image src="https://bucket.region.amazonaws.com/path/to/file.jpg" {...others} />
```

Getting:

```
Bad request

INVALID_IMAGE_OPTIMIZE_REQUEST
```

when deploying on vercel.

Any idea? 

Thanks!

**Comment by grndctrl** [comment]
I ran into the same wall, but I found out I had my `images.domains` missconfigured. Fixing that, solved it for me, so maybe double check the actual domain where you image is hosted?

**Reply by mk0y** [reply]
What's wrong with domain name here? It's correct. I'm also having this problem and can't solve it. I'm loading images from Contentful using `ctfassets.net` as domain but doesn't work.

**Reply by Ivaylo-Korakov** [reply]
@grndctrl is one of the most useless comments :) If you copy the domain he gave and ctrl+F it on the page you will immediately see that the domain is configured correctly. Right here your comment becomes pointless. Also, the person who asked the question clearly stated that HE DID THAT CORRECTLY and he is asking for help since THE THING YOU ARE SUGGESTING IS NOT WORKING. Why would you even point out a thing which is a copy-paste from the first line of the documentation about the Image tag in next js. We can all read the documentation just like he did and configured it correctly. He opened an issue for after he checked the docs.....

**Reply by CGaul** [reply]
I had the same `Bad request INVALID_IMAGE_OPTIMIZE_REQUEST` issue, while loading an Asset Image from Contentful and solved it by implementing a [loader for the Next image](https://nextjs.org/docs/api-reference/next/image#loader) instead of injecting a URL into the src argument.

So the following code resulted in the `Bad request` error:
```javascript
import Image from "next/future/image"
import { Block, BLOCKS, Inline } from "@contentful/rich-text-types"

const contentfulRichTextRenderOptions = {
  renderNode: {
    [BLOCKS.EMBEDDED_ASSET]: (node: Block | Inline) => {
      return (
        <Image
          src={`https://${node.data.target.fields.file.url}`}
          height={node.data.target.fields.file.details.image.height}
          width={node.data.target.fields.file.details.image.width}
          alt={node.data.target.fields.description}
        />
      )
    },
  },
}
```

this returns a HTML img (again, which fails with the above 400 BAD REQUEST):
```html
<img alt="" srcset="/_next/image?url=https%3A%2F%2F%2F%2Fimages.ctfassets.net..." src="/_next/image?url=https%3A%2F%2F%2F%2Fimages.ctfassets.net..." decoding="async" data-nimg="future" loading="lazy" style="color:transparent" width="2095" height="715">
```

while a loader implementation worked for me:
```javascript
import { Block, BLOCKS, Inline } from "@contentful/rich-text-types"
import Image, { ImageLoaderProps } from "next/future/image"
import { ImageLoader } from "next/dist/client/future/image"


const contentfulImageLoader: ImageLoader = ({ src, width }: ImageLoaderProps) => {
  return `${src}?w=${width}`
}

const contentfulRichTextRenderOptions = {
  renderNode: {
    [BLOCKS.EMBEDDED_ASSET]: (node: Block | Inline) => {
      return (
        <Image
          loader={contentfulImageLoader}
          src={node.data.target.fields.file.url}
          height={node.data.target.fields.file.details.image.height}
          width={node.data.target.fields.file.details.image.width}
          alt={node.data.target.fields.description}
        />
      )
    },
  },
}
```
as stated in the comment below, this turns of next image optimization.

Thus it returns a HTML img without the /_next/image part:

```html
<img alt="" srcset="//images.ctfassets.net/..." src="//images.ctfassets.net/..." decoding="async" data-nimg="future" loading="lazy" style="color:transparent;width:100%;height:auto;margin:1rem 0" width="2095" height="715">
```

on `next.config.js` I have the following configuration for contentful (I think however, domains is still enough to configure):
```javascript
images: {
      domains: ["images.ctfassets.net"],
      remotePatterns: [
        {
          protocol: "https",
          hostname: "images.ctfassets.net",
        },
      ],
    },
```

hope this helps someone, too!

**Reply by rubengmurray** [reply]
Yep, this is the answer. It's just a bit weird why locally this isn't an issue.

I added the following for some Instagram images from their API and it fixed it immediately.

```
    remotePatterns: [
      {
        protocol: 'https',
        hostname: '**.cdninstagram.com',
      },
    ],
```

After a small amount of reading, I do think this is fairly well covered in the docs.

1. https://nextjs.org/docs/api-reference/next/image#src
2. https://nextjs.org/docs/api-reference/next/image#remote-patterns

**Reply by hassangulzar1** [reply]
I ran into the same error bro when I deployed the project on Vercel then contentful images not showing can you have any solution to resolve this?

**Reply by Jyrko** [reply]
@CGaul Thanks, used approach with loader, still works. Next.js v15.0.1

**Comment by willslater** [comment]
Also having this issue? Any more information on how you resolved? Works fine locally.

**Reply by Vadman97** [reply]
This turned out to be a capitalization issue for me. I was using images from my /public folder with a .JPG extension. Renaming the files to .jpg fixed the error once on the Vercel hosted site.

**Comment by veronica-severyn** [comment]
Same here.. Works fine locally. But there's an issue once on Vercel. NextJS 11.1.2

**Comment by halchester** [comment]
In next.config.js file, you have to add the domain of your aws s3 bucket. For me, it's directly streamed from Express so I just need to add the backend URL to the config file.

**Reply by willslater** [reply]
Yeah, I realised I was using qa s3 bucket address stupidly, I think I also had issue from leaving / at the end so maybe try that if anyone experiencing the same.

**Comment by AryanJ-NYC** [comment]
I'm also having this problem now. The image renders just fine locally but craps out when on Vercel.

You can see the error on this page: https://scarce.city/marketplace/digital/QRRARE

It tries to fetch: https://scarce.city/_next/image?url=https%3A%2F%2FscannableNFTs.com%2Fart%2Fthumb%2FQRRARE.gif&w=828&q=50 which results in `INVALID_IMAGE_OPTIMIZE_REQUEST`.

My next.config:

```typescript
const config = {
  images: {
    domains: [
      'arweave.net',
      'bitcorns.com',
      'cdn.bitcoincoaster.com',
      'cdn.sanity.io',
      'dankdirectory.files.wordpress.com',
      'fakeraredirectory.com',
      'fakeraredirectory.files.wordpress.com',
      'hosting.photobucket.com',
      'ipfs.io',
      'i.imgur.com',
      'mandelduck.herokuapp.com',
      'metadata.spellsofgenesis.com',
      'pepewtf.s3.amazonaws.com',
      'phunchkins.com',
      'public-scarce-city.s3.amazonaws.com',
      'rarejapanesenfts.com',
      'rarepepedirectory.com',
      'rarepepewallet.com',
      'scannablenfts.com', // <--- here it is
      'thewojakway.com',
      'xchain.io',
      'xcp.dankdirectory.site',
      'xcp.dankinfo.art',
      'xcpinata.s3.amazonaws.com',
    ],
  },
  env: {
    // my env
  },
};
```

**Reply by daveidivide** [reply]
What was your fix for this?

Everything seems fine on your site, so I assumeyou got it working?

**Reply by AryanJ-NYC** [reply]
It ended up being mismatching casing:

https://github.com/vercel/next.js/discussions/20138#discussioncomment-3073510

**Comment by AryanJ-NYC** [comment]
Fun fact: the domains array is case sensitive so if you're trying to render an image from `aBc.com` but have `abc.com` in your domains array, it won't work.

**Reply by thmarinho** [reply]
Well you saved me a few hours...

**Comment by michalniem** [comment]
Any luck with this? I'm getting `INVALID_IMAGE_OPTIMIZE_REQUEST` on my arweave.net remote images. It's not consistent, I'm getting a 400 for random images, but the rest are working fine. Next: 12.2.0

**Reply by willslater** [reply]
Is there anything to notice about filenaming? As in, do the ones that fail follow a different naming convention? Do the same ones always fail? 

I haven’t had this issue for a while but I remember naming and leaving / in the domain caused me the issues.

**Reply by michalniem** [reply]
Ah, forgot to mention, that all images work locally for me. It's deployed into vercel. No diffs between those files regarding name convention.
This is working example: $domainName/image?url=https%3A%2F%2Farweave.net%2FHQavlBC_XQrOQ3RC_HN5_-BquDJ4LlBNmvcXt_8zfjM&w=384&q=75
This is NOT working example: $domainName/_next/image?url=https%3A%2F%2Farweave.net%2FWWs7dvunVtEtHyCO_HJanD5YpXCyuwd01EpypF3gDgA&w=384&q=75

**Reply by AryanJ-NYC** [reply]
Arweave is a rough one. Times out a ton for me so hard to diagnose.

**Reply by MirekR** [reply]
We think we have figured out when this happens. It seems like cache issue on the nextjs and vercel. If on the first request to arweave does not return image (for example when you try to requested before it's fully propagated on the arweave network), vercel/nextjs still cached "broken" response and until cache expires, it keeps returning 400 INVALID_IMAGE_OPTIMIZE_REQUEST  response. 

You can verify it, when you have broken link, add cache buster param to the url (like ```%3F_%3D123``` => ```$domainName/_next/image?url=https%3A%2F%2Farweave.net%2FWWs7dvunVtEtHyCO_HJanD5YpXCyuwd01EpypF3gDgA%3F_%3D123&w=384&q=75```
) and you should the image fine.

**Reply by f0-x** [reply]
Dear Mr @MirekR could you please explain how to implement this? I couldn't figure out how to do this, anyone?

**Reply by this-is-simon** [reply]
> Dear Mr @MirekR could you please explain how to implement this? I couldn't figure out how to do this, anyone?

What @MirekR is suggesting is: You can add characters to the end of the image's url and it will "cache bust" the erroring image. For example if you have on the front end:
`const imageToDisplay = data.urlImage`
You can do:
`const imageToDisplay = `${data.urlImage}?_=my-random-string``
Or more dynamically:
`const imageToDisplay = `${data.urlImage}?_=${Date.now()}``

It effectively removes caching from next/Image and makes it very expensive (in monetary terms) to retrieve images from vercel.

**Comment by jonasmarco** [comment]
Any Update?

**Comment by qruz-hq** [comment]
Having the exact same issue, but only on vercel, and no way of getting a hand on the error log for some reason

**Comment by f-deniyi** [comment]
If you are still having the error after setting up your next config correctly then try this, create a function` const loaderProp = ({ src }) => {
    return src
  } ` then pass this function to the loader props of Image like `<Image loader={loaderProp}
/>`

**Reply by Eerkz** [reply]
this only turns off the the Image optimization.

**Reply by Coyas** [reply]
it  works for me, but without optimizations its not the way

**Comment by riccardolardi** [comment]
I'm still having this issue - any update?

**Comment by ghost** [comment]
any updates on this issue? Same here

**Reply by FlugHeli** [reply]
I am new to Next.js and i have the same problem

**Comment by laszlo-horvath** [comment]
Update in 2023 June: I had the same problem for 1 particular image.

I ended up using the [unoptimized](https://nextjs.org/docs/pages/api-reference/components/image#unoptimized) prop on the Next `<Image />` tag and it fixed the issue for me by disabling the [Image Optimization](https://nextjs.org/docs/pages/building-your-application/optimizing/images) for that image only.

**Reply by tdsilve** [reply]
This works for me

**Reply by TOcvfan** [reply]
me too

**Reply by wbcdev** [reply]
Thank you.  This worked for my google maps places implementation.

**Comment by mertburakd** [comment]
any update on this issuse?

**Comment by 4D5A90** [comment]
Same problem but returning an image from Route Handlers (look like this in production https://mydomain.vercel.app/_next/image?url=/api/images/1)

**Reply by arxk** [reply]
Same thing happening to me with an internal image route.

**Reply by erickreutz** [reply]
I'm also having this problem.

**Reply by fvermaut** [reply]
Same for me.  To "solve" it I added the "unoptimized" prop on the Image component.   Not ideal though.

**Reply by ebulku** [reply]
I am using Route Handlers to pass image and fixed by adding the full url like this:


```
import { headers } from 'next/headers'
//////
const headersList = await headers()
const proto = headersList.get('x-forwarded-proto')
const host = headersList.get('host')
const fullHost = `${proto}://${host}`
//////
<Image
    width={500}
    height={500}
    src={`${fullHost}/api/files/${item.id}`}
    alt=""
  />

```

And add these in `remotePatterns` in `next.config.ts`

```
{
  protocol: 'https',
  hostname: '**.vercel.app',
  port: '',
  pathname: '**',
},
{
  protocol: 'http',
  hostname: 'localhost',
  port: '3000',
  pathname: '**',
},

```

**Comment by stcaterings** [comment]
same +1

**Comment by AbhayaShankar** [comment]
I am also facing the same Issue, any update on this ?

**Comment by KyrillGobber** [comment]
I deployed on vercel and had this issue. just adding "**.vercel.app" as hostname in remotePatters worked to solve this:
```
const nextConfig = {
    images: {
        remotePatterns: [
            {
                protocol: 'https',
                hostname: '**.vercel.app',
            },
            {
                protocol: 'http',
                hostname: 'localhost',
                port: '3000',
                pathname: '/api/media/**',
            },
        ],
    }
}
```

**Comment by LaxAce** [comment]
I was able to fix this by ensuring that the Images folder in my `public` folder is started with a capital letter `I` so the file is named `Images` and not `images`. So it read as follows `<Image src="/Images/logo.png" width={140} height={50} alt="Logo" />`

**Comment by sajjadhussain14** [comment]
I had the same issue with external images not optimizing in Next.js! After much debugging, here's the solution that worked for me:"

🚨 The Problem
tsx
Copy
// This breaks in production (Vercel)
<Image
  src="https://www.abc.com/prodimages/123.jpg"
  alt="Product"
  width={300}
  height={300}
/>
❌ Error: 400: INVALID_IMAGE_OPTIMIZE_REQUEST
❌ Why: Next.js blocks unconfigured external domains

🛠 The Fix
1️⃣ First, whitelist your domain in next.config.js:

javascript
Copy
module.exports = {
  images: {
    remotePatterns: [
      { 
        protocol: 'https',
        hostname: 'abc.com', // ← Your domain here
      },
    ],
  },
}
2️⃣ Then, add a loader to handle URL formatting:

tsx
Copy
const customLoader = ({ src, width, quality }) => {
  return `${src}?w=${width}&q=${quality || 75}`
}

// Usage:
<Image
  loader={customLoader}
  src="https://www.abc.com/prodimages/123.jpg"
  alt="Product"
  width={300}
  height={300}
/>
✅ Why This Works
remotePatterns tells Next.js which domains to optimize

loader ensures proper URL structure for the image optimizer

Works perfectly on Vercel and other hosts

**Comment by shakerscode** [comment]
Guys, I got the same error and my PM was asking to resolve the issue. Initially I thought, it is a `next.config` issue. But it was not. Then I had to search in google and reach here. I tried some of the answers, but they didn't work for me. Here's what works for me:  

Before going there, lets learn why this error happen: 

```
This is a common issue when using Vercel's Image Optimization feature. Here’s a clear breakdown of what's happening and how to fix it.

🎯 The Direct Answer: What OPTIMIZED_IMAGE_REQUEST_PAYMENT_REQUIRED Means

This error is not a problem with your React code or your next.config.js.

The error Status 402 Payment Required is coming directly from Vercel. It means that your Vercel account has exceeded the free Image Optimization quota allowed for your plan (likely the Hobby/free plan).

Vercel's Hobby plan includes a specific number of optimized images per month (e.g., 1,000). Every time next/image requests a new size (w=750&q=75 in your URL), it counts as one optimization. Once you hit that limit, Vercel stops processing new images and returns this 402 error until your billing cycle resets or you upgrade. [Answer is collected from an AI]
```
 
**### Now to resolve the issue, let's follow these 2 way:**
**1.** 
`<Image
      src={src}
      alt="your_alt" 
      ...
      unoptimized={true} // <-- Add this line
    />`

**2.**

```
i. Check Your Vercel Usage 

First, confirm this is the issue by checking your dashboard:

-     Go to your Vercel project.
-     Click on the Usage tab.
-     Scroll down to the Image Optimization section. You will likely see that you have used 100% (or more) of your included quota.


ii. Wait or Upgrade Your Plan

-     Wait: If this is a personal project, the simplest "fix" is to wait for your Vercel billing cycle to reset (usually at the start of the month).
-     Upgrade: If this is a production app with real users, you should upgrade to a Vercel Pro plan. The Pro plan gives you a much-Higher limit and allows you to pay for overages, so your images will never break for users.

    ```

I will be happy if this helps. Thanks guys.
